stl/inc/xstring

```diff
@@ -161,11 +161,17 @@ struct _Char_traits { // properties of a string or stream element
     }
 
     static _CONSTEXPR20 _Elem* assign(
-        _Out_writes_all_(_Count) _Elem* const _First, size_t _Count, const _Elem _Ch) noexcept
-    /* strengthened */ {
+        _Out_writes_all_(_Count) _Elem* const _First, size_t _Count, const _Elem _Ch) noexcept /* strengthened */ {
         // assign _Count * _Ch to [_First, ...)
         for (_Elem* _Next = _First; _Count > 0; --_Count, ++_Next) {
-            *_Next = _Ch;
+#ifdef __cpp_lib_constexpr_string
+            if (_STD is_constant_evaluated()) {
+                _STD construct_at(_Next, _Ch);
+            } else
+#endif // __cpp_lib_constexpr_string
+            {
+                *_Next = _Ch;
+            }

```

Comment:
Note we should also apply the special casing for the assign(left, right) method below